squidDrone/src/i2c/i2c_interface.hpp

```diff
@@ -0,0 +1,47 @@
+#ifndef SRC_I2C_INTERFACE_HPP_
+#define SRC_I2C_INTERFACE_HPP_
+
+#include "i2c_status.hpp"
+#include <stdint.h>
+#include <vector>
+#include <tuple>
+
+namespace i2c {
+
+  /**
+   * @brief Interface of I2C package
+   * 
+   */
+  class I2CInterface {
+    public:
+      virtual ~I2CInterface() = default;
+      explicit I2CInterface(void){};
+
+      /**
+       * @brief Reads data from I2C Bus
+       * 
+       * @param address The address of the I2C Bus participant
+       * @param byte_size Amount of Bytes to read
+       * @param timeout Timeout in milliseconds
+       * @return std::tuple<I2CStatus, std::vector<uint8_t>> Status of I2C Interface and Data read as std::vector. Each element is a Byte read.
+       */
+      virtual auto Read(uint8_t address, uint16_t byte_size, uint32_t timeout=I2C_STANDARD_TIMEOUT_IN_MS) noexcept -> std::tuple<I2CStatus, std::vector<uint8_t>> = 0;
+      
+      /**
+       * @brief Writes data to I2C participant
+       * 
+       * @param address The address of the I2C Bus participant
```

Comment:
What is the range of addresses that you are testing against?

Reply:
For the user it doesn't matter, that I test against valid I2C addresses. Every Chip normally has one or two addresses "burnt" in. So the User has to use them. But I test against a valid range for Chips to use, taken from some I2C documentation

Reply:
ok, but you still have to write down the valid range for your public interface.

Reply:
I already added some ranges in a later commit: See here: https://github.com/elheck/Squiddrone/blob/e578c51813e066b04af12f4a910e87de8dfaf400/squidDrone/src/i2c/i2c_interface.hpp#L25